### PR TITLE
Ali/populate table

### LINKED ISSUE #2: Have JSON data populate table (sort by date)
Should be a singular table where data is sorted by date with most recent date at the top

### LINKED ISSUE #9: Make table responsive
Figure out how to style table for mobile view. Only table is within scope, everything else is for #4

### PR DESCRIPTION
**Pull request by a3alani**
## Tracking Info

Resolves #2

## Changes

- I added a JS function in app.js that returns the tasks in order sorted by date.
- In index.html, I displayed the tasks in sorted order in a table. Note for that I used the code in script tags inside the HTML file, but the JS function could also be used by calling it and it will return the tasks ordered by date if needed.

## Testing

- I checked the return of the JS function by logging the output on the console and comparing the results to the expected output. It indeed returned the tasks in order sorted by date.

## Confirmation of Change

>UPDATED
Screenshot 2024-04-23 at 3 19 44 PM

>OLD

Screenshot 2024-04-21 at 7 18 50 PM

### CODE REVIEW

**Comment by advaithm22** [comment]
I think the "Date" feature should be displayed in "month/day/year,  time" format, similar to what Edward did. Here is an example: 
Screen Shot 2024-04-22 at 11 50 02 AM

File: app.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        parsed_dates.push({
+          id_val: task.id,
+          title_val: task.title,
+          descriptions_val: task.descriptions,
+          date_val: new Date(Date.parse(task.due_date)),
+          completed_val: task.completed,
+        });
+      });
+      parsed_dates.sort((a, b) => a.date_val - b.date_val);
+      console.log("sorted parsed dates", parsed_dates);
```
Comment (brentonmdunn) [inline app.js:29]: Can you remove the `console.log`?

File: app.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+/**
+ * function iterates over tasks from JSON file
+ * Sortes the tasks by due date
+ * returns an array of JSON objects sorted by due date
+ * 
+ return data type example: [{},{},{},{},{},{}]
```
Comment (brentonmdunn) [inline app.js:6]: Probably keep the header on top but remove the sample json data below

File: index.html [hunk 3]
```diff
@@ -1,13 +1,62 @@
...
+    <link rel="stylesheet" href="styles.css" />
     <title>Document</title>
-</head>
-<body>
+  </head>
+  <body>
+    <table>
+      <thead>
+        <tr>
+          <!-- <th>ID</th>
```
Comment (brentonmdunn) [inline index.html:13]: remove comments

File: index.html [hunk 4]
```diff
@@ -1,13 +1,62 @@
...
\ No newline at end of file
+    <script>
+      const parsed_dates = [];
+      fetch("tasks.json")
+        .then((response) => response.json())
+        .then((data) => {
+          data.forEach((task) => {
+            parsed_dates.push({
+              // id_val: task.id,
+              // title_val: task.title,
```
Comment (brentonmdunn) [inline index.html:31]: remove comments (I know I was the one that commented it out but I just wanted to make sure I didn't break anything)

File: index.html [hunk 5]
```diff
@@ -1,13 +1,62 @@
...
+              descriptions_val: task.description,
+              date_val: new Date(Date.parse(task.due_date)),
+              completed_val: task.completed,
+            });
+          });
+          parsed_dates.sort((a, b) => a.date_val - b.date_val);
+          console.log("sorted parsed dates", parsed_dates);
```
Comment (brentonmdunn) [inline index.html:38]: remove `console.log`

File: index.html [hunk 6]
```diff
@@ -1,13 +1,62 @@
...
-<body>
+  </head>
+  <body>
+    <table>
+      <thead>
+        <tr>
+          <!-- <th>ID</th>
+          <th>Title</th> -->
+          <th>Task</th>
+          <th>Due Date</th>
```
Comment (brentonmdunn) [inline index.html:16]: Can you flip the columns so that due date is column on the left and task is column on the right?

**Review by brentonmdunn — CHANGES_REQUESTED** [review]
Just a few small changes, otherwise looks great

**Review by advaithm22 — APPROVED** [review]

**Pull request merged**